client/src/main/java/com/paypal/selion/platform/grid/LocalSelendroidNode.java

```diff
@@ -15,191 +15,168 @@
 
 package com.paypal.selion.platform.grid;
 
-import io.selendroid.grid.SelendroidSessionProxy;
-import io.selendroid.standalone.SelendroidConfiguration;
-import io.selendroid.standalone.SelendroidLauncher;
-
-import java.util.ArrayList;
-import java.util.List;
-import java.util.logging.Handler;
-import java.util.logging.Level;
-import java.util.logging.Logger;
-
 import org.apache.commons.configuration.ConversionException;
-import org.openqa.grid.common.exception.GridException;
+import org.apache.commons.lang.StringUtils;
+import org.openqa.selenium.net.NetworkUtils;
+import org.openqa.selenium.net.PortProber;
 
+import com.google.common.annotations.Beta;
 import com.paypal.selion.configuration.Config;
 import com.paypal.selion.configuration.Config.ConfigProperty;
+import com.paypal.selion.grid.ProcessLauncherOptions;
+import com.paypal.selion.grid.SelendroidJarSpawner;
 import com.paypal.selion.logger.SeLionLogger;
 import com.paypal.test.utilities.logging.SimpleLogger;
 
 /**
- * A singleton that is responsible for encapsulating all the logic w.r.t starting/shutting down a local android driver
- * node.
- * 
+ * A singleton that is responsible for encapsulating all the logic w.r.t starting/shutting down a local selendroid node.
  */
-class LocalSelendroidNode extends AbstractNode implements LocalServerComponent {
+@Beta
+final class LocalSelendroidNode extends AbstractBaseLocalServerComponent implements LocalServerComponent {
+    private static final SimpleLogger LOGGER = SeLionLogger.getLogger();
+    private static volatile LocalSelendroidNode instance;
+
+    static synchronized final LocalServerComponent getSingleton() {
```

Comment:
It exists for two reasons;
* So an `@Test` method which relys on `LocalGridManager` to boot the 'LocalSelendroidNode' instance, can access the bootede `LocalSelendroidNode` instance
* To override the implementation of `getSingleton()` in `AbstractBaseLocalServerComponent` which throws an `IllegalStateException` (by design).